zilliqa/src/pool.rs

```diff
@@ -256,6 +256,27 @@ impl TransactionPool {
     pub fn size(&self) -> usize {
         self.transactions.len()
     }
+
+    pub fn has_txn_for_next_block(&self) -> bool {
+        let mut ready = self.ready.clone();
+        while let Some(ReadyItem { tx_index, .. }) = ready.pop() {
+            // We don't include nonceless txns because the way we present results on API level requires having proper nonce
+            if let TxIndex::Intershard(_, _) = tx_index {
+                continue;
+            }
```

Comment:
Are you sure this applies in this function? I would imagine that if there is an `Intershard` transaction in the pool, it would be ready for the next block and we should return true.

Reply:
I agree, we should include intershard txns asap and not wait for other txns to be ready for the next block